internal/pkg/deploy/cloudformation/project.go

```diff
@@ -449,48 +449,73 @@ func (cf CloudFormation) waitForStackSetOperation(stackSetName, operationID stri
 }
 
 // DeleteProject deletes all project specific StackSet and Stack resources.
-func (cf CloudFormation) DeleteProject(name string, accounts, regions []string) error {
-	stackSetName := fmt.Sprintf("%s-infrastructure", name)
+func (cf CloudFormation) DeleteProject(projectName string) error {
+	stackSetName := fmt.Sprintf("%s-infrastructure", projectName)
+	if err := cf.deleteProjectStackSet(stackSetName); err != nil {
+		return err
+	}
+	return cf.delete(fmt.Sprintf("%s-infrastructure-roles", projectName))
+}
 
-	if _, err := cf.client.DeleteStackInstances(&cloudformation.DeleteStackInstancesInput{
-		Accounts:     aws.StringSlice(accounts),
-		RetainStacks: aws.Bool(false),
-		Regions:      aws.StringSlice(regions),
+func (cf CloudFormation) deleteProjectStackSet(stackSetName string) error {
+	stackInstances, err := cf.client.ListStackInstances(&cloudformation.ListStackInstancesInput{
 		StackSetName: aws.String(stackSetName),
-	}); err != nil {
-		return fmt.Errorf("DeleteStackInstances for stackset %s, accounts %s, and regions %s: %w",
-			stackSetName, accounts, regions, err)
+	})
+
+	if err != nil {
+		// If the stackset doesn't exist - just move on.
+		if stackSetDoesNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("fetching existing project stack instances: %w", err)
+	}
+
+	// We want to delete all the stack instances, so we create
+	// a set of account ids and regions.
+	accountSet := map[string]bool{}
+	regionSet := map[string]bool{}
+	for _, summary := range stackInstances.Summaries {
+		accountSet[*summary.Account] = true
+		regionSet[*summary.Region] = true
 	}
 
-	// The looping here is because there's no Wait convenience function the SDK provides
-	// for waiting on StackSet Instances to delete. The DeleteStackSet call will return the
-	// ErrCodeOperationInProgressException while instances are still actively being deleted.
-	maxAttempts := maxDeleteStackSetAttempts
-	for maxAttempts > 0 {
-		_, err := cf.client.DeleteStackSet(&cloudformation.DeleteStackSetInput{
+	regions := []string{}
+	accounts := []string{}
+	for key := range accountSet {
```

Comment:
Good question! So the way stack sets work is that it will create a stack instance for every possible Region x Account pair, so order doesn't matter. In our case, we only have one account, the project account :)